### PR TITLE
cargo-download: init at 0.1.1

### PR DESCRIPTION
**Pull request by alyssais**
Cargo.nix was generated from cargo-download 0.1.1 with this patch applied, which was necessary to not confuse carnix about the version being different between Cargo.toml and Cargo.lock.

See https://github.com/Xion/cargo-download/pull/3.

```diff
diff --git a/Cargo.lock b/Cargo.lock
index cc7357c..c384dc1 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -76,7 +76,7 @@ dependencies = [

 [[package]]
 name = "cargo-download"
-version = "0.1.0"
+version = "0.1.1"
 dependencies = [
  "ansi_term 0.9.0 (registry+https://github.com/rust-lang/crates.io-index)",
  "clap 2.27.1 (registry+https://github.com/rust-lang/crates.io-index)",
```

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/package-management/cargo-download/default.nix [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{ lib, buildPlatform, buildRustCrate, fetchgit }:
+
+((import ./Cargo.nix { inherit lib buildPlatform buildRustCrate fetchgit; }).cargo_download {}).override {
+
+}
```
Comment (xeji) [inline pkgs/tools/package-management/cargo-download/default.nix]: Why the empty override?

Comment (xeji) [reply pkgs/tools/package-management/cargo-download/default.nix]: Nevermind, I see that's what the examples in the manual are structured like...

**Comment by xeji** [comment]
@GrahamcOfBorg  build cargo-download

**Comment by xeji** [comment]
Some missing frameworks for Darwin. Unless you have a Mac this may be a few rounds of trial-and-error with ofborg because Security depends on other frameworks etc.  You can try to fix it or just limit platforms to linux. Up to you.

**Comment by xeji** [comment]
@GrahamcOfBorg build cargo-download

File: pkgs/tools/package-management/cargo-download/default.nix [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{ stdenv, lib, fetchgit, darwin, buildPlatform
+, buildRustCrate, defaultCrateOverrides }:
+
+((import ./Cargo.nix { inherit lib buildPlatform buildRustCrate fetchgit; }).cargo_download {}).override {
+  crateOverrides = defaultCrateOverrides // {
+    cargo-download = attrs: {
+      buildInputs = lib.optional stdenv.isDarwin
```
Comment (alyssais) [inline pkgs/tools/package-management/cargo-download/default.nix]: I noticed that cargo-edit used a propagatedBuildInput for Security. I wasn't sure if I should be doing the same here. Should I be?

Comment (xeji) [reply pkgs/tools/package-management/cargo-download/default.nix]: Sorry, don't know too much about how the Darwin frameworks work.
If it works for you on a Mac it should be ok.

**Pull request merged**